.github/workflows/update_runtime_deps.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+
+"""Update upper bounds of runtime dependencies in pyproject.toml using UV."""
+
+import subprocess
+import sys
+import tempfile
+
+import toml
+from packaging.requirements import Requirement
+from packaging.specifiers import Specifier, SpecifierSet
+
+
+def write_unbounded_pyproject(pyproject_file: str, workdir: str) -> None:
+    """Write pyproject.toml without upper dependency bounds to a temp directory."""
+    pyproject = toml.load(pyproject_file)
+    old_deps = [Requirement(dep) for dep in pyproject["project"]["dependencies"]]
+
+    new_deps = []
+    for req in old_deps:
+        req.specifier = SpecifierSet(
+            [spec for spec in req.specifier if spec.operator != "<="]
+        )
+        new_deps.append(str(req))
+    pyproject["project"]["dependencies"] = new_deps
+
+    # Write modified pyproject.toml to temp directory
+    temp_pyproject_path = f"{workdir}/pyproject.toml"
+    with open(temp_pyproject_path, "w") as f:
+        toml.dump(pyproject, f)
+
+
+def update_requirements(tmpdir: str) -> str:
+    """Use UV to resolve and bump all dependencies of a pyproject.toml in tmpdir.
+
+    Returns the resolved environment as a requirements.txt string.
+    """
+    res = subprocess.run(
+        ["uv", "export", "--no-hashes", "--refresh"],
+        cwd=tmpdir,
+        capture_output=True,
+        text=True,
+    )
+
+    if res.returncode != 0:
+        print("Error running UV to resolve dependencies:", res.stderr)
+        sys.exit(1)
+
+    resolved_env = res.stdout
+    return resolved_env
+
+
+def get_updated_bounds(pyproject_file: str, resolved_env: str) -> list[str]:
+    """Parse resolved environment and update upper bounds from original pyproject.toml."""
+    pyproject = toml.load(pyproject_file)
+    current_deps = [Requirement(dep) for dep in pyproject["project"]["dependencies"]]
+
+    new_upper_bounds = {}
+    for line in resolved_env.splitlines():
+        if "==" in line:
+            req = Requirement(line)
+            pkg = req.name
+            ver = next(iter(req.specifier)).version
+
+            if pkg in new_upper_bounds:
+                if ver > new_upper_bounds[pkg]:
```

Comment:
Subtle bug here I think; for instance
```python
>>> "10.0.0" > "9.0.0"
False
```
Using something like `Version(ver) > Version(new_upper_bounds[pkg])` (from `packaging.version`) would be better